tools/lkl/lib/virtio.c

```diff
@@ -544,8 +545,11 @@ int virtio_dev_setup(struct virtio_dev *dev, int queues, int num_max)
 	}
 
 	avail = sizeof(lkl_virtio_devs) - (devs - lkl_virtio_devs);
-	devs += snprintf(devs, avail, " virtio_mmio.device=%d@0x%lx:%d",
+	num_bytes = snprintf(devs, avail, " virtio_mmio.device=%d@0x%lx:%d",
 			 mmio_size, (uintptr_t)dev->base, dev->irq);
+	if (num_bytes < 0 || num_bytes >= avail)
+		return -LKL_ENOMEM;
```

Comment:
We need to cleanup before exiting in case of error: unregister_iomem(), free(dev->queue), etc.